# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/nr_socket_tcp.h
# Commit: c43589fd0f90
# Full Hash: c43589fd0f90d174eb185b563406fd0a2e6ea712
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Update OnConnected callbacks to specify what type of proxying (if any) is being used, and implement IsProxied based on this information. r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45186
# ==============================================================================

diff -r 940d7f787bea -r c43589fd0f90 media/mtransport/nr_socket_tcp.h
--- a/media/mtransport/nr_socket_tcp.h	Tue Sep 17 18:18:15 2019 +0000
+++ b/media/mtransport/nr_socket_tcp.h	Tue Sep 17 18:18:33 2019 +0000
@@ -86,11 +86,13 @@
   int listen(int aBacklog) override;
   int accept(nr_transport_addr* aAddr, nr_socket** aSocket) override;
 
-  bool IsProxied() const override { return true; }
+  bool IsProxied() const override {
+    return mProxyType != "" && mProxyType != "direct";
+  }
 
   // WebrtcTCPSocketCallback
   void OnClose(nsresult aReason) override;
-  void OnConnected() override;
+  void OnConnected(const nsCString& aProxyType) override;
   void OnRead(nsTArray<uint8_t>&& aReadData) override;
 
   size_t CountUnreadBytes() const;
@@ -107,6 +109,7 @@
   bool mClosed;
 
   size_t mReadOffset;
+  nsCString mProxyType;
   std::list<NrTcpSocketData> mReadQueue;
 
   std::shared_ptr<NrSocketProxyConfig> mConfig;